pkgs/top-level/all-packages.nix

```
@@ -23433,7 +23433,7 @@ in
 
   netlogo = callPackage ../applications/science/misc/netlogo { };
 
-  ns-3 = callPackage ../development/libraries/science/networking/ns3 { };
+  ns-3 = callPackage ../development/libraries/science/networking/ns-3 { python = python3; };
```

or change directory name
```suggestion
  ns-3 = callPackage ../development/libraries/science/networking/ns3 { python = python3; };
```